pilot/pkg/model/service.go

```diff
@@ -60,10 +60,35 @@ type Service struct {
 	// ServiceAccounts specifies the service accounts that run the service.
 	ServiceAccounts []string `json:"serviceaccounts,omitempty"`
 
+	// MeshExternal (if true) indicates that the service is external to the mesh.
+	// These services are defined using Istio's ExternalService spec.
+	MeshExternal bool
+
 	// LoadBalancingDisabled indicates that no load balancing should be done for this service.
 	LoadBalancingDisabled bool `json:"-"`
+
+	// Resolution indicates how the service instances need to be resolved before routing
+	// traffic. Most services in the service registry will use static load balancing wherein
+	// the proxy will decide the service instance that will receive the traffic. External services
+	// could either use DNS load balancing (i.e. proxy will query DNS server for the IP of the service)
+	// or use the passthrough model (i.e. proxy will forward the traffic to the network endpoint requested
+	// by the caller)
+	Resolution Resolution
```

Comment:
Are `Resolution` values other than `ClientSideLB` (i.e. 0) applicable for internal services? If not, would it make sense to combine the `MeshExternal` field with `Resolution`, or maybe just delete `MeshExternal`?

Reply:
Headless services in k8s shouldn't be load balanced by Istio, so they would use `Passthrough` resolution.